ext/rugged/rugged.c

```diff
@@ -111,6 +111,28 @@ static VALUE rb_git_features(VALUE self)
 
 /*
  *  call-seq:
+ *    Rugged.valid_full_oid?(oid) -> true or false
+ *
+ *  Checks to see if a string contains a full 40-character sha1.
+ *
+ *    Rugged.valid_full_oid?('d8786bfc97485e8d7b19b21fb88c8ef1f199fc3f')
+ *    #=> true
+ */
+static VALUE rb_git_valid_full_oid(VALUE self, VALUE hex)
+{
+	git_oid oid;
+
+	Check_Type(hex, T_STRING);
+	int errorcode = git_oid_fromstr(&oid, StringValueCStr(hex));
```

Comment:
Yes, but Microsoft went through a long period of not giving a shit about C, and they didn't bother to include support for this until VS 2013.  Sorry!